### ISSUE TITLE
Add Frechet RandomVariable

### ISSUE THREAD

**Issue by bwengals** [aesara-devs/aesara#1240]
## Description of your problem or feature request

I'd like to request to add a Frechet random variable implementation, parameterized the same as [Stan here](https://mc-stan.org/docs/functions-reference/frechet-distribution.html).  It's another name for the inverse Weibull, which is in [scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.invweibull.html).  The use case is penalized complexity (PC) priors for lengthscale parameters in Gaussian process models.  

So I'm proposing this as an enhancement, and would be happy to make a PR for it.  Would like to support it in aeppl too.

**Comment by ricardoV94** [comment]
You could make it a simple function wrapper that just returns `at.inv(weibull(...))`. Aeppl will soon be able to derive the logprob as well: https://github.com/aesara-devs/aeppl/pull/144

**Comment by brandonwillard** [comment]
We've converged on the notion of only implementing the core functionality and `numpy.random.*` implementations in this repository, so we might need to move this to AePPL or actually start working on a separate, larger Aesara SciPy project.

**Comment by bwengals** [comment]
Thanks for answering.

@ricardoV94 you're saying Aeppl will derive the logprob of the inverse Weibull from the logprob of the Weibull?  Very cool!

@brandonwillard Whichever of those two choices sounds good to me.  I just subclassed `ScipyRandomVariable`, so it shouldn't be too tricky to add, or I can file it away until there's an Aesara Scipy thing.

**Comment by ricardoV94** [comment]
> @ricardoV94 you're saying Aeppl will derive the logprob of the inverse Weibull from the logprob of the Weibull?  Very cool!

Yes. 

But I also suggested there might be no good reason to implement a specific Op for the Frechet, unless the scipy RNG is doing something fancy that is not achieved by just taking draws from a Weibull and inverting them.

**Comment by brandonwillard** [comment]
> But I also suggested there might be no good reason to implement a specific Op for the Frechet, unless the scipy RNG is doing something fancy that is not achieved by just taking draws from a Weibull and inverting them.

Yes, that will always be the preferred route in cases like this, but, as you mention, we might not get complete parity with NumPy&mdash;at least in terms of performance and/or seeding and sample results.  Such parity concerns are one of the reasons I don't want to go too far in our `scipy.stats` coverage, because, at some point along that path, we're simply rolling our own implementations and that requires considerably more testing.

**Comment by rlouf** [comment]
When it comes to probability distribution my vote goes to AePPL.

Also `at.inv(weibull)` is nice but we should still have a way to give those a type so we can reason about them in AeMCMC.

**Comment by ricardoV94** [comment]
> Also at.inv(weibull) is nice but we should still have a way to give those a type so we can reason about them in AeMCMC.

What do you mean by giving them a type? It should be trivial to pattern match such graph. 

Anyway it's the same argument as always for not adding more Ops if not need (no need to worry about backends, test as thoroughly, etc...)

**Comment by rlouf** [comment]
Would you say we should drop `LogNormal`, since it's just an exponentiated `Normal`? And `Binomial`? And `InvGamma`?

The choice of what we name and what we leave as function of another RV is somewhat arbitrary, and I just wanted to point out that we don't have a criterion to decide yet.

We can either decide that whatever statisticians named they named for a reason, and get the largest set of names function possible. Or decide to only keep a subset of fundamental distributions from which everything else can be derived. These two propositions are not incompatible, as we can always give a name to e.g. `at.inv(weibull)` to keep the smallest subset of logprobs/forward sampling functions.

**Comment by rlouf** [comment]
> I think the criteria will ultimately be: Does the specialized Op produce significantly faster/reliable draws (or logp graphs) than the derived one?

I agree for the draws and the logprob part. But is there a way we could define an `Op` such that `at.inv(weibull)` is used when conditioning on this RV? (that's vague, but I hope you get the idea). That's something that could be done in AePPL, where we should have a wrapper for RVs implemented in Aesara (and add other RVs) anyway.

**Comment by ricardoV94** [comment]
> Is there a way we could define an Op such that at.inv(weibull) is used when conditioning on this RV?

What do you mean?

**Comment by rlouf** [comment]
Implement a `Frechet` Op, say in AePPL, so that the following graph:

```python
x_rv = aeppl.random.Frechet(1, 1, 1)
```

is equivalent to:

```python
x_rv = at.inv(at.random.weibull(1, 1))
```

as far as sampling and logprob are concerned. An `OpFromGraph` of some sort, I suppose.

**Comment by ricardoV94** [comment]
You mean the case where you implement a specialized `RandomVariable` `Op` but don't want to write a `logprob` function by hand? I don't see how that would be possible.

Otherwise the following would work without any special logic after https://github.com/aesara-devs/aeppl/pull/144:
```python
x_rv = at.inv(at.random.weibull(1, 1))
x_vv = x_rv.type()

joint_logprob({x_rv: x_vv})
```

No need to use a `FunctionGraph` or anything (at least as far as aeppl is concerned).
